src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaInductionPersistenceAdapter.kt

```diff
@@ -17,8 +17,10 @@ class JpaInductionPersistenceAdapter(
 
   @Transactional
   override fun createInduction(createInductionDto: CreateInductionDto): Induction {
-    val persistedEntity = inductionRepository.save(inductionMapper.fromCreateDtoToEntity(createInductionDto))
-    return inductionMapper.fromEntityToDomain(persistedEntity)
+    inductionRepository.save(inductionMapper.fromCreateDtoToEntity(createInductionDto))
+
+    // return the persisted entity with the JPA fields populated
+    return getInduction(createInductionDto.prisonNumber)!!
```

Comment:
Previously the returned entity didn't have the JPA generated fields populated, but calling the separate `getInduction()` method in this class gets around this (though I'm pretty sure this has nothing to do with transaction boundaries for reasons I can explain separately).